File: Projects/Framework-Samples/PLINQO/CSharp/Tracker/Source/Tracker.Tests/ClearCacheTests.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web;
using CodeSmith.Data.Linq;
using NUnit.Framework;
using Tracker.Data;
using System.Data.Linq;

namespace Tracker.Tests.CacheTests
{
    [TestFixture]
    public class ClearCacheTests : RoleTests
    {
        [Test]
        public void SimpleTest()
        {
            try
            {
                using (var db = new TrackerDataContext())
                {
                    var query = db.Role.Where(r => r.Name == "Test Role");
                    var key = query.GetKey();
                    var roles = query.FromCache();

                    var cache1 = HttpRuntime.Cache.Get(key);
                    Assert.IsNotNull(cache1);
                    Assert.AreSame(roles, cache1);

                    query.ClearCache();

                    var cache2 = HttpRuntime.Cache.Get(key);
                    Assert.IsNull(cache2);
                }
            }
            catch (AssertionException)
            {
                throw;
            }
            catch
            {
                Assert.Fail();
            }
        }
    }
}

File: Projects/Framework-Samples/PLINQO/VB/Tracker/Source/Tracker.Tests/CacheTests/ClearCacheTests.vb
Imports System.Linq
Imports System.Web
Imports CodeSmith.Data.Linq
Imports NUnit.Framework
Imports Tracker.Core.Data

Namespace Tracker.Tests.CacheTests
    <TestFixture()> _
    Public Class ClearCacheTests
        Inherits RoleTests
        <Test()> _
        Public Sub SimpleTest()
            Try
                Using db = New TrackerDataContext()
                    Dim query = db.Role.Where(Function(r) r.Name = "Test Role")
                    Dim key = query.GetKey()
                    Dim roles = query.FromCache()

                    Dim cache1 = HttpRuntime.Cache.[Get](key)
                    Assert.IsNotNull(cache1)
                    Assert.AreSame(roles, cache1)

                    query.ClearCache()

                    Dim cache2 = HttpRuntime.Cache.[Get](key)
                    Assert.IsNull(cache2)
                End Using
            Catch generatedExceptionName As AssertionException
                Throw
            Catch
                Assert.Fail()
            End Try
        End Sub
    End Class
End Namespace
